src/Rules/MissingTypehintCheck.php

```diff
@@ -71,6 +74,15 @@ public function getIterableTypesWithMissingValueTypehint(Type $type): array
 			if ($type instanceof AccessoryType) {
 				return $type;
 			}
+			if ($type instanceof ConditionalType || $type instanceof ConditionalTypeForParameter) {
+				$iterablesWithMissingValueTypehint = array_merge(
+					$iterablesWithMissingValueTypehint,
+					$this->getIterableTypesWithMissingValueTypehint($type->getIf()),
+					$this->getIterableTypesWithMissingValueTypehint($type->getElse()),
+				);
+
+				return $type;
+			}
```

Comment:
Not overly excited by this. An alternative would be to get the resolved type, but that might skip over types that disappear after normalization.